rollup.config.js

```diff
@@ -1,37 +1,66 @@
 const commonjs = require('rollup-plugin-commonjs');
-const path = require('path');
 const resolve = require('rollup-plugin-node-resolve');
 const typescript = require('rollup-plugin-typescript');
 const pkg = require('./package.json');
 
-const selfId = path.resolve(__dirname, 'src/iife-self-placeholder.js');
+const globals = {
+  'whatwg-fetch': 'FakeFetch',
+  'fake-xml-http-request': 'FakeXMLHttpRequest',
+  'route-recognizer': 'RouteRecognizer',
+};
+
+const iife = {
+  banner:
+`var Pretender = (function (self) {
+
+  function getModuleDefault(module) {
+    return module.default || module;	
+  }
+
+  var appearsBrowserified = typeof self !== 'undefined' &&
+                            typeof process !== 'undefined' &&
+                            (Object.prototype.toString.call(process) === '[object Object]' ||
+                             Object.prototype.toString.call(process) === '[object process]');
+
+  var RouteRecognizer = appearsBrowserified ? getModuleDefault(require('route-recognizer')) : self.RouteRecognizer;
+  var FakeXMLHttpRequest = appearsBrowserified ? getModuleDefault(require('fake-xml-http-request')) :
+    self.FakeXMLHttpRequest;
+
+  // fetch related ponyfills
+  var FakeFetch = appearsBrowserified ? getModuleDefault(require('whatwg-fetch')) : self.WHATWGFetch;
+`,
+  footer:
+`
+if (typeof module === 'object') {
+  module.exports = Pretender;	
+} else if (typeof define !== 'undefined') {	
+  define('pretender', [], function() {	
+    return Pretender;	
+  });
+}
+
+self.Pretender = Pretender;
+
+return Pretender;
+
+}(self));`,
+};
 
 module.exports = {
   input: 'src/index.ts',
-  external: [
-    selfId,
-    'whatwg-fetch',
-    'fake-xml-http-request',
-    'route-recognizer',
-  ],
+  external: Object.keys(pkg.dependencies),
   output: [
     {
       name: 'Pretender',
       file: pkg.main,
       format: 'iife',
-      globals: {
-        [selfId]: 'self',
-        'whatwg-fetch': 'FakeFetch',
-        'fake-xml-http-request': 'FakeXMLHttpRequest',
-        'route-recognizer': 'RouteRecognizer',
-      },
-      banner: 'var FakeFetch = self.WHATWGFetch;\n' +
-              'var FakeXMLHttpRequest = self.FakeXMLHttpRequest;\n' +
-              'var RouteRecognizer = self.RouteRecognizer;\n',
+      globals: globals,
+      banner: iife.banner,
+      footer: iife.footer,
```

Comment:
I'm using an object to extract the iife's header and footer to reduce the noise in the rollup config.

Reply:
through refactoring the banner/footer into a JS file we just read and split at a special comment, this ends up being cleaned up even more!